### PR TITLE
txn: get txn status does not block if txn not found

### PR DESCRIPTION
**Pull request by cfzjywxk**
### What problem does this PR solve? 

For the `Rollback` statement from user, the pessimistic locks will be rollbacked and the primary key in store has no related information. There are possibilities that some  other transactions do `checkTxnStatus` on these locks  and they will be blocked `ttl` time

### What is changed and how it works?
 
Let the transaction do retry pessimistic lock if `txn not found` and the lock does not expire yet

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

Release note

### CODE REVIEW

File: session/pessimistic_test.go [hunk 1]
```diff
@@ -1096,3 +1096,40 @@ func (s *testPessimisticSuite) TestBatchPointGetLockIndex(c *C) {
...
+		var err error
+		defer func() {
+			errCh <- err
+		}()
+		_, err = tk2.Exec("update t1 set c3 = 100 where c1 = 1")
+		if err != nil {
+			return
+		}
+	}()
+	time.Sleep(time.Millisecond * 30)
```
Comment (coocood) [inline session/pessimistic_test.go]: Is the test is unstable?

Comment (cfzjywxk) [reply session/pessimistic_test.go]: A better way is to use some `sync-point` like utilities, let `tk2` pause before `getTxnStatusFromLock`, and `tk` rollback the transaction, then tk2 continues execution, for now seems sleep is a simple way

**Comment by coocood** [comment]
LGTM

**Comment by cfzjywxk** [comment]
@jackysp @lysu @tiancaiamao PTAL

**Comment by lysu** [comment]
so after "prewrite primary key first", it seems we have no other situations that will see a old txn's primary key no longer exist except this "self-rollback"?

**Comment by cfzjywxk** [comment]
> so after "prewrite primary key first", it seems we have no other situations that will see a old txn's primary key no longer exist except this "self-rollback"?

In some situations with cluster exceptions, this may still happen. But after this change, the blocked transactions will not just retry `checkTxnStatus` but also retry `pessimistic lock`, so it will not be blocked for `ttl` time even `pk not found` happens

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**